bigframes/session/__init__.py

```diff
@@ -1506,8 +1508,14 @@ def _start_query(
         """
         Starts query job and waits for results.
         """
+        api_methods = log_adapter._api_methods
         job_config = self._prepare_job_config(job_config)
+        job_config.labels = bigframes_io.create_job_configs_labels(
+            job_configs_labels=job_config.labels, api_methods=api_methods
+        )
         query_job = self.bqclient.query(sql, job_config=job_config)
+        # Clear out the global api logger
+        log_adapter._api_methods = []
```

Comment:
This all needs to be protected by a lock. Alternatively, let's make a `log_adapter.get_and_reset_api_methods()` that locks correctly and returns a copy of the previous list of api methods.